indra_gpt/processors/post_processors.py

```diff
@@ -0,0 +1,175 @@
+            )
+        # Ensure pmids are available and correctly sized
+        if input_pmids and len(input_pmids) != len(input_texts):
+            self.logger.warning(
+                f"PMID count ({len(input_pmids)}) does not match input text count ({len(input_texts)}). "
+                "Proceeding without PMIDs where missing."
+            )
+
+        processed_stmts_json_list = []
+        for i, response in enumerate(generated_responses):
+            text = input_texts[i]
+            pmid = input_pmids[i] if input_pmids else "Not provided"
+            try:
+                # 1. Convert to json object if it is a string
+                if isinstance(response, str):
+                    try:
+                        stmt_json_response = json.loads(response)
+                    except json.JSONDecodeError as e:
+                        raise ValueError("Invalid JSON string provided.") from e
+                else: 
+                    stmt_json_response = response
+                # 2. If the JSON object contains a list of statements under the "statements" key,
+                # retrieve the value. This is currently specifically for the case
+                # of openai chat completions for structured output mode using a specific schema. 
+                if stmt_json_response.get("statements"):
+                    stmts_json = stmt_json_response["statements"]
+                else:
+                    if isinstance(stmt_json_response, list):
+                        stmts_json = stmt_json_response
+                    else:
+                        stmts_json = [stmt_json_response]
+                # 3. Map the stmt class names to the correct class names 
+                stmts_json = [self.map_stmt_class_name(stmt_json) for stmt_json in stmts_json]
+                # 4. Remove fields with null values
+                stmts_json = [self.remove_empty_strings_and_lists(stmt_json) for stmt_json in stmts_json]
+                # 5. Update evidence since the generated statements likely have dummy values for evidence
+                stmts_json = [self.update_evidence(stmt_json, text, pmid, self.config) for stmt_json in stmts_json]
+                processed_stmts_json_list.append(stmts_json)
+            except Exception as e:
+                self.logger.error(f"Error processing response: {response} | Error: {e}")
+                processed_stmts_json_list.append({"error": str(e)})
+
+        preassembled_statements = [self.preassemble_pipeline(stmt) for stmt in processed_stmts_json_list]
+        self.logger.info(f"Completed post-processing: {len(preassembled_statements)} statements processed.")
+        return preassembled_statements
+    
+    def map_stmt_class_name(self, stmt_json):
+        stmt_classes = get_all_descendants(Statement)
+        stmt_mapping = {stmt_class.__name__.lower(): stmt_class.__name__ for stmt_class in stmt_classes}
```

Comment:
`stmt_classes` and `stmt_mapping` should be at the module level instead of in here.